### DISCUSSION TITLE
[데이터 공유] User group 데이터

### DISCUSSION THREAD

**Discussion by ddoyoon** [bcaitech1/p4-dkt-lastffang#131] [Show and tell]
### User group 나눈 방법
- User 벡터 생성: assessment_category 9 가지에 대해 푼 적이 있으면 1, 없으면 0으로 하는 벡터로 user profile 을 생성
- `sklearn.cluster.AgglomerativeClustering` 으로 유저 벡터들을 유사도 기반으로 9 그룹으로 나눔 -> `user_group` column
![image](https://user-images.githubusercontent.com/18363932/121881977-e64cd100-cd4a-11eb-91d5-53f83dbb85a1.png)
- 각 user_group 내에서 평균 정답률을 중심으로 한번 더 그룹화
### 파일 설명
- 드라이브 링크
- ug_orig.zip: raw train, test data 전체에 대해 피쳐 추가한 데이터
- ug_filtered.zip: ug_orig.zip 의 데이터를 testId 기반으로 필터링
### Column 설명
- `user_group`: assessment_category 기반 유저 그룹
- `user_group_rated`: user_group 평균 정답률보다 정답률이 높다면 1, 낮다면 0 을 user_group 값 뒤에 concat 함
    - e.g. user_group 7 번에서 정답률이 평균보다 낮은 유저는 '70' 이 됩니다
    - 이렇게 표현한 값 자체에 큰 의미는 없고 groupby 를 위해 나눴습니다..! 
- `user_group_answer_mean`: **user_group_rated** 의 평균 정답률
- `user_group_time_log_mean`: **user_group_rated** 의 log(소요시간) 의 평균 
- `user_group_time_mean`: **user_group_rated** 의 raw 소요시간의 평균
- 건희, 근영님이 효과를 보신 `time_lda` 와 `KnowledgeTagGroup_time_mean` 도 추가되어있으나, testId 기반 필터링 전에 전체 데이터 대상으로 계산한 값이라 값이 완전 동일하지는 않습니다!
    - 동일한 값으로 사용하시려면 필터링된 ug_filtered 에서 해당 column 들을 다시 계산해주셔야 합니다!
### 실험 결과
- `user_group_time_mean` 과 `user_group_time_log_mean` 중에서 로그를 취한 후자가 성능이 살짝 더 나은 것 같았고 여기에 추가로 `user_group_answer_mean` 을 써서 리더보드에 제출했습니다.
    - 실험 로그를 자세히 보시려면 wandb run name "dy-gruattn-ug" 를 검색해주세요
- Validation 기준 avg_auc 가 0.824 정도였는데, 제출하니 public 기준 acc 는 오르고 auc 는 떨어졌습니다 ㅠㅠ
![image](https://user-images.githubusercontent.com/18363932/121883608-e5b53a00-cd4c-11eb-82f5-464414646daf.png)

**Comment by GY-Jeong** [comment]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 저거 세개는 acc가 올라갈수록 auc가 떨어지네요ㅠㅠㅠㅠㅠㅠㅠㅠ

**Reply by ddoyoon** [reply]
ㅋㅋㅋㅋ 그러게요 ㅠㅠ answer 관련된 피쳐를 넣은게 혹시 오히려 독이 된걸까요..? 차이가 다 고만고만해서 리더보드에 내기 전에 판단이 어렵네요 😭
